Sources/Controllers/DashboardOnMap/MapSettings/MapSettingsGpxViewController.swift

```diff
@@ -422,21 +462,17 @@ final class MapSettingsGpxViewController: OABaseNavbarSubviewViewController {
     }
     
     private func updateSelectedRows() {
-        guard let visibleGpxFilePaths = OAAppSettings.sharedManager()?.mapSettingVisibleGpx.get() else { return }
         let gpxListToShow = isSearchActive ? filteredGpxList : (isShowingVisibleTracks ? visibleGpxList : allGpxList)
-        if !isSearchFilteringActive {
-            selectedGpxTracks.removeAll()
-            for gpx in gpxListToShow where visibleGpxFilePaths.contains(gpx.gpxFilePath) {
-                selectedGpxTracks.append(gpx)
-            }
-        }
-        
-        for (index, gpx) in gpxListToShow.enumerated() {
-            let indexPath = IndexPath(row: index, section: 0)
-            if selectedGpxTracks.contains(where: { $0.gpxFilePath == gpx.gpxFilePath }) {
-                tableView.selectRow(at: indexPath, animated: false, scrollPosition: .none)
-            } else {
-                tableView.deselectRow(at: indexPath, animated: false)
+        let recentlyVisible = isShowingVisibleTracks ? recentlyVisibleGpxList : []
+        let sections = [gpxListToShow, recentlyVisible]
+        for (sectionIndex, section) in sections.enumerated() {
```

Comment:
for (sectionIndex, section) in [gpxListToShow, recentlyVisibleTracks].enumerated() {